nixos/modules/services/networking/headscale.nix

```diff
@@ -462,6 +462,22 @@ in {
       isSystemUser = true;
     };
 
+    services.nginx = {
```

Comment:
I want to call out that as a project, headscale is not recommending to put proxies in front as odd behaviour might occur.

That said, I am running it with a proxy, and it is more of a "if you dont get it right, you might run into strange and weird behaviour", so I am more positive to giving everyone the same tried and tested config than having people concoct up their own stuff.

I would also say that if we add nginx options in the module, it should cover other "easy to mess up" scenarios like gRPC and potentially metrics, here is how I do it for my config:
https://github.com/kradalby/dotfiles/blob/master/machines/headscale.oracldn/headscale.nix#L164-L169